packages/react-native-tracer-provider/src/EmbraceNativeSpan.ts

```diff
@@ -0,0 +1,224 @@
+   *  - return a blank span context
+   *
+   * Behaviour can be figured on the EmbraceNativeTracerProvider by supplying spanContextSyncBehaviour
+   */
+  public spanContext(): SpanContext {
+    if (this.savedSpanContext) {
+      return this.savedSpanContext;
+    }
+
+    switch (this.spanContextSyncBehaviour) {
+      case 'block':
+        return TracerProviderModule.spanContext(this.nativeID());
+      case 'throw':
+        throw new Error('span context is not yet available on the native side');
+      case 'return_empty':
+        return {
+          traceId: '',
+          spanId: '',
+          traceFlags: 0,
+        };
+    }
+  }
+
+  /**
+   * spanContextAsync provides an async alternative to spanContext by wrapping the response in a promise
+   */
+  public spanContextAsync(): Promise<SpanContext> {
+    return this.creating || Promise.reject('failed to retrieve span context');
+  }
+
+  public setAttribute(key: string, value: AttributeValue): this {
+    return this.setAttributes({ [key]: value });
+  }
+
+  public setAttributes(attributes: Attributes): this {
+    if (this.isReadonly()) {
+      return this;
+    }
+    TracerProviderModule.setAttributes(this.nativeID(), normalizeAttributes(attributes));
+    return this;
+  }
+
+  public addEvent(
+    name: string,
+    attributesOrStartTime?: Attributes | TimeInput,
+    timeStamp?: TimeInput
+  ): this {
+    if (this.isReadonly()) {
+      return this;
+    }
+
+    if (isAttributes(attributesOrStartTime)) {
+      TracerProviderModule.addEvent(
+        this.nativeID(),
+        name,
+        normalizeAttributes(attributesOrStartTime),
+        normalizeTime(timeStamp)
+      );
+    } else {
+      TracerProviderModule.addEvent(
+        this.nativeID(),
+        name,
+        normalizeAttributes({}),
+        normalizeTime(attributesOrStartTime)
+      );
+    }
+
+    return this;
+  }
+
+  public addLink(link: Link): this {
+    return this.addLinks([link]);
+  }
+
+  public addLinks(links: Link[]): this {
+    if (this.isReadonly()) {
+      return this;
+    }
+
+    TracerProviderModule.addLinks(this.nativeID(), normalizeLinks(links));
+
+    return this;
+  }
+
+  public setStatus(status: SpanStatus): this {
+    if (this.isReadonly()) {
+      return this;
+    }
+
+    TracerProviderModule.setStatus(this.nativeID(), {
+      code: SpanStatusCode[status.code],
+      message: status.message,
+    });
+
+    return this;
+  }
+
+  public updateName(name: string): this {
+    if (this.isReadonly()) {
+      return this;
+    }
+
+    TracerProviderModule.updateName(this.nativeID(), name);
+
+    return this;
+  }
+
+  public end(endTime?: TimeInput): void {
+    if (this.isReadonly()) {
+      return;
+    }
+
+    TracerProviderModule.endSpan(this.nativeID(), normalizeTime(endTime));
+    this.recording = false;
+  }
+
+  public isRecording(): boolean {
+    return this.recording;
+  }
+
+  public recordException(exception: Exception, time?: TimeInput): void {
```

Comment:
Is there a reason this one doesn't return `this` like some of the ones above it?

Reply:
this is driven by the OTEL JS API so it's forced to follow this interface: https://github.com/open-telemetry/opentelemetry-js/blob/main/api/src/trace/span.ts#L149